herd/tests/instructions/AArch64.MTE/L05.litmus

```diff
@@ -0,0 +1,16 @@
+AArch64 L05
```

Comment:
It would be really good if herd7 could ignore the size of STZG for L03 but throw an error for L05. But I am not sure this is possible. So I guess you're suggesting that STZG should require mixed mode don't you?

Reply:
You mean throw an error when the target location of `STZG` is an array in non-mixed mode?

Reply:
I think so yes. I think it would be useful to allow herd7 to work without enabling `-variant mixed` in cases like `L03` after all, `L03` is not incorrect (the user didn't check `x[1]` they're only checking `x` aka `x[0]`). But `L05` is wrong so they can't trust the result and they should enable `-variant mixed`. What do you think? Is it possible?

Reply:
Your solution implies looking at several places in the execution: (1) apply `STZG` to an array location and (2) reading some array cell different from the first.

I'd favor a solution that would look at the `STZG` write only. Then if we can check that STZG base address is an array address, we can do this only when the address is knows (ie  is read from initial state)
1. Fail with a message that suggests using `-variant mixed`, or
2. Fill all the array cells with zero, (up to granule size ?) 

Also, if the address is not known 1. also apply.

A more simple but less expressive solution would be to fail, still suggesting `-variant mixed` when we both have some array and `STZG` instruction in the same test.

Reply:
> A more simple but less expressive solution would be to fail, still suggesting -variant mixed when we both have some array and STZG instruction in the same test.

I think that's a very good middle ground. But in any case don't worry too much if it's too hard we can fix in a subsequent PR.

Reply:
I have finally implemented this middle ground, as it is relatively straightforward.